### PR TITLE
feat: Update Ktor to Ktor 2.0.0 + Finished uploadAsset API

### PR DESCRIPTION
**Pull request by gongracr**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Upload assets API was broken before and we were still using Ktor `1.6.7`.

### Solutions
Bumped to Ktor 2.0.0 version and updated all the API methods making use of post and get methods of the legacy bundle.

### Testing

Just run the ConversationsApplication.kt class

### CODE REVIEW

**Review by vitorhugods — APPROVED** [review]
Really nice 🎉, just two tiny comments.

File: cli/src/main/kotlin/com/wire/kalium/cli/ConversationsApplication.kt [hunk 1]
```diff
@@ -40,9 +45,37 @@ class ConversationsApplication : CliktCommand() {
...
+        val imageBytes: ByteArray? = getResource("moon1.jpg")
+        val uploadResult = networkModule.assetApi.uploadAsset(
+            AssetMetadata("image/jpeg", true, AssetRetentionType.ETERNAL, calcMd5(imageBytes)),
+            imageBytes ?: ByteArray(16)
+        )
+        println("The upload result is -> $uploadResult")
+    }
+
+    @Throws(IOException::class)
+    fun getResource(name: String?): ByteArray? {
```
Comment (MohamadJaara) [inline cli/src/main/kotlin/com/wire/kalium/cli/ConversationsApplication.kt:63]: is there any reason why name is nullable here ?

File: cli/src/main/kotlin/com/wire/kalium/cli/ConversationsApplication.kt [hunk 2]
```diff
@@ -40,9 +45,37 @@ class ConversationsApplication : CliktCommand() {
...
+        val classLoader = this::class.java.classLoader
+        classLoader.getResourceAsStream(name).use { resourceAsStream -> return toByteArray(resourceAsStream) }
+    }
+
+    @Throws(IOException::class)
+    fun toByteArray(input: InputStream?): ByteArray? = input?.let { inputStream ->
```
Comment (MohamadJaara) [inline cli/src/main/kotlin/com/wire/kalium/cli/ConversationsApplication.kt:69]: if this code is from legacy then it was interpolated from java and in our usecase no need for it to be nullable

**Review by MohamadJaara — COMMENTED** [review]
great job 🥳, just small comments

File: cli/src/main/kotlin/com/wire/kalium/cli/ConversationsApplication.kt (hunk 1 above)
Comment (gongracr) [reply cli/src/main/kotlin/com/wire/kalium/cli/ConversationsApplication.kt:63]: Just copied from the legacy API class

File: cli/src/main/kotlin/com/wire/kalium/cli/ConversationsApplication.kt (hunk 2 above)
Comment (gongracr) [reply cli/src/main/kotlin/com/wire/kalium/cli/ConversationsApplication.kt:69]: Correct. I removed the nullable type

**Pull request merged**